sequence_processing_pipeline/FastQCJob.py

```diff
@@ -94,13 +95,33 @@ def _find_projects(self, path_to_run_id_data_fastq_dir, is_raw_input):
                      'zero_files' not in x]
 
             # break files up into R1, R2, I1, I2
-            # assume _R1_ does not occur in the path as well.
-            r1_only = [x for x in files if '_R1_' in x]
-            r2_only = [x for x in files if '_R2_' in x]
-
-            # amplicon runs may or may not have an i2. this is okay.
-            i1_only = [x for x in files if '_I1_' in x]
-            i2_only = [x for x in files if '_I2_' in x]
+            # use capturing to handle both raw files as well as trimmed and
+            # filtered files. We don't need to process the captured string.
+            i1_files = compile(r"^.*_L\d{3}_I1_\d{3}\.(trimmed\.|filtered"
```

Comment:
Why the capture group, it doesn't seem like it's used?

Reply:
A side of effect of using the capture group is that I can use one regex to handle '.trimmed.fastq.gz',  '.filtered.fastq.gz', and  '.fastq.gz' files. Brackets ('[',']') and ORs ('|') didn't work for me.

Reply:
It is not good practice to rely on side effects... Suggest adding `?:`, see

```python
>>> import re
>>> re.compile(r'(?:foo|bar)?baz').match('baz')
<re.Match object; span=(0, 3), match='baz'>
>>> re.compile(r'(?:foo|bar)?baz').match('foobaz')
<re.Match object; span=(0, 6), match='foobaz'>
>>> re.compile(r'(?:foo|bar)?baz').match('barbaz')
<re.Match object; span=(0, 6), match='barbaz'>
>>> re.compile(r'(?:foo|bar)?baz').match('fbaz')
>>>
```